src/plugins/legacy/medExportVideo/medExportVideo.cpp

```diff
@@ -0,0 +1,425 @@
+            if (image.isNull())
+            {
+                d->imagesArray.remove(cpt);
+            }
+            else
+            {
+                cpt++;
+            }
+        }
+
+        // Run export video/jpeg
+        if (displayFileDialog() == medAbstractProcessLegacy::SUCCESS)
+        {
+            QApplication::setOverrideCursor(Qt::WaitCursor);
+            QApplication::processEvents();
+            qDebug() << metaObject()->className() <<" ENCODING... w h "<<d->width<<"/"<<d->height;
+
+            int res = medAbstractProcessLegacy::FAILURE;
+
+            if (d->format == JPGBATCH)
+            {
+                res = this->exportAsJPEG();
+            }
+            else
+            {
+                res = this->exportAsVideo();
+            }
+
+            qDebug() << metaObject()->className() <<" END OF ENCODING -- "<<res;
+            QApplication::restoreOverrideCursor();
+
+            return res;
+        }
+    }
+    return medAbstractProcessLegacy::FAILURE;
+}
+
+int ExportVideo::exportAsJPEG()
+{
+    // Get the index of images base name
+    int lastPoint = d->filename.size();
+    if (d->filename.contains("."))
+    {
+        lastPoint = d->filename.lastIndexOf(".");
+    }
+    if (d->filename.at(lastPoint-1) == QString::number(0).at(0))
+    {
+        lastPoint = lastPoint - 1;
+    }
+
+    int cpt = 0;
+    for (QImage qimage : d->imagesArray)
+    {
+        // Qt to VTK images (vtkImageData)
+        vtkSmartPointer<vtkQImageToImageSource> qimageToImageSource = vtkSmartPointer<vtkQImageToImageSource>::New();
+        qimageToImageSource->SetQImage(&qimage);
+        qimageToImageSource->Update();
+
+        // Current image filename
+        QString name = d->filename.left(lastPoint) + QString::number(cpt)+".jpg";
+
+        vtkSmartPointer<vtkJPEGWriter> writerJPEG = vtkSmartPointer<vtkJPEGWriter>::New();
+        writerJPEG->SetFileName(name.toStdString().c_str());
+        writerJPEG->SetInputConnection(qimageToImageSource->GetOutputPort());
+        writerJPEG->Write();
+
+        cpt++;
+    }
+
+    return medAbstractProcessLegacy::SUCCESS;
+}
+
+int ExportVideo::exportAsVideo()
+{
+    vtkSmartPointer<vtkImageCanvasSource2D> source = vtkSmartPointer<vtkImageCanvasSource2D>::New();
+    source->SetScalarTypeToUnsignedChar();
+    source->SetNumberOfScalarComponents(3);
+    source->SetExtent(0, d->width-1, 0, d->height-1, 0, 0);
+
+    vtkSmartPointer<vtkGenericMovieWriter> writerVideo;
+
+    if (d->format == OGGVORBIS)
+    {
+        vtkSmartPointer<vtkOggTheoraWriter> writerVideoTmp = vtkSmartPointer<vtkOggTheoraWriter>::New();
+        writerVideoTmp->SetInputConnection(source->GetOutputPort());
+        writerVideoTmp->SetFileName(d->filename.toStdString().c_str());
+        writerVideoTmp->SetRate(d->frameRate);
+        writerVideoTmp->SetSubsampling(d->subsampling);
+        writerVideoTmp->SetQuality(d->quality);
+
+        writerVideo = writerVideoTmp;
+    }
+    else
+    {
```

Comment:
Be careful!
What appends if entre in with:

- d->format with other value than FFMPEG
- d->format at FFMPEG but with MED_USE_FFmpeg not define

suggestion:
 
`#ifdef (MED_USE_FFmpeg)`
`        else if (d->format == FFMPEG)`
`        {`
`             vtkSmartPointer<vtkFFMPEGWriter> writerVideoTmp = vtkSmartPointer<vtkFFMPEGWriter>::New();`
`             writerVideoTmp->SetInputConnection(source->GetOutputPort());`
`             writerVideoTmp->SetFileName(d->filename.toStdString().c_str());`
`             writerVideoTmp->SetRate(d->frameRate);`
`             writerVideoTmp->SetQuality(d->quality);`
`             writerVideo = writerVideoTmp;`
`        }`
`#endif`
`        else`
`        {`
`            return medAbstractProcessLegacy::FAILURE;`
`        }`
`    }`